sql/testdata/builtin_function

```diff
@@ -258,13 +258,75 @@ query error syntax error at or near "\("
 SELECT current_date()
 
 query T
-SELECT now() - current_timestamp < interval '10s'
+SELECT now() - transaction_date::timestamp < interval '24h10s'
+----
+true
+
+query error syntax error at or near "\("
+SELECT transaction_date()
+
+query T
+SELECT now() - current_timestamp = interval '0s'
 ----
 true
 
 query error syntax error at or near "\("
 SELECT current_timestamp()
 
+query error syntax error at or near "\("
+SELECT statement_timestamp()
+
+query T
+SELECT now() - statement_timestamp = interval '0s'
+----
+true
+
+query T
+SELECT statement_timestamp - statement_timestamp = interval '0s'
+----
+true
+
+query error syntax error at or near "\("
+SELECT clock_timestamp()
```

Comment:
No strong reason, we should support all functions like unique_id() without the parenthesis. It's nice to not have the parenthesis, making it look like a variable in SQL. I'll allow the variable use as a function.